src/update.c

```diff
@@ -555,6 +555,15 @@ int main_update()
 	run_scripts();
 	grabtime_stop(&times);
 
+	/* Create the state file that will tell swupd it's on a mix on future runs */
+	if (mix_exists && !system_on_mix()) {
+		int fd = open("/usr/share/defaults/swupd/mixed", O_RDWR | O_CREAT, S_IRWXU | S_IRGRP | S_IROTH);
```

Comment:
the executable bit should not be set, so you can change `S_IRWXU` to `S_IRUSR | S_IWUSR`